src/app/components/blogs/blogs.component.tsx

```diff
@@ -1,33 +1,32 @@
 import React from 'react';
 import { useStoreon } from 'storeon/react';
-import { Blog } from '../../modules/content/content.state';
+import { useEffect } from 'react';
+import { LoadingIndicator } from '../loadingIndicator/loadingIndicator.component';
+import { ContentLabel } from '../../modules/content/content.state';
+import { CategoryItems } from '../categoryItems.component.tsx/categoryItems.component';
 import './blogs.component.scss';
-import {
-    ContentLabel,
-    ContentTile,
-} from '../contentItem/contentTile.component';
 
 export function Blogs() {
     const { content } = useStoreon('content');
 
-    function renderBlogs(): JSX.Element {
-        if (!content || !content.blogs.length) {
-            return <div>No data</div>;
-        }
+    useEffect(() => {}, [content.pending]);
 
-        return content.blogs.map((blog: Blog, index: number) => (
-            <ContentTile
-                key={index + blog.title}
-                contentItem={blog}
-                contentLabel={ContentLabel.Blog}
-            />
-        ));
+    if (content.pending) {
+        return <LoadingIndicator />;
     }
-
-    return (
-        <div className="blogs-component">
-            <h2>Blogs</h2>
-            {renderBlogs()}
-        </div>
-    );
+    if (content.blogs.length) {
+        return (
+            content && (
+                <div className="blogs-content">
+                    <h2>Blogs</h2>
+                    <CategoryItems
+                        categoryItems={content.blogs}
+                        label={ContentLabel.Blog}
+                        specifiedLength={500}
+                    />
+                </div>
+            )
+        );
+    }
+    return <div></div>;
```

Comment:
do not return empty div, add content.error handling

Reply:
Refactored, as mentioned in another comment.